includes/Context.php

```diff
@@ -348,27 +352,7 @@ public function get_amp_mode() {
 				&& $exposes_support_mode;
 		}
 
-		if ( $exposes_support_mode ) {
-			// If recent version, we can properly detect the mode.
-			if ( $amp_plugin_version_2_or_higher ) {
-				$mode = AMP_Options_Manager::get_option( 'theme_support' );
-			} else {
-				$mode = AMP_Theme_Support::get_support_mode();
-			}
-
-			if ( AMP_Theme_Support::STANDARD_MODE_SLUG === $mode ) {
-				return self::AMP_MODE_PRIMARY;
-			}
-
-			if ( in_array( $mode, array( AMP_Theme_Support::TRANSITIONAL_MODE_SLUG, AMP_Theme_Support::READER_MODE_SLUG ), true ) ) {
-				return self::AMP_MODE_SECONDARY;
-			}
-		} elseif ( function_exists( 'amp_is_canonical' ) ) {
-			// On older versions, if it is not primary AMP, it is definitely secondary AMP (transitional or reader mode).
-			if ( amp_is_canonical() ) {
-				return self::AMP_MODE_PRIMARY;
-			}
-
+		if ( $exposes_support_mode || function_exists( 'amp_is_canonical' ) ) {
 			return self::AMP_MODE_SECONDARY;
 		}
```

Comment:
Let's keep the original check in here, but add `STANDARD_MODE_SLUG` in, eg:

```php
if ( $exposes_support_mode ) {
    // If recent version, we can properly detect the mode.
    if ( $amp_plugin_version_2_or_higher ) {
        $mode = AMP_Options_Manager::get_option( 'theme_support' );
    } else {
        $mode = AMP_Theme_Support::get_support_mode();
    }

    if (
        in_array( $mode, array(
            AMP_Theme_Support::STANDARD_MODE_SLUG,
            AMP_Theme_Support::TRANSITIONAL_MODE_SLUG,
            AMP_Theme_Support::READER_MODE_SLUG
        ), true )
    ) {
        return self::AMP_MODE_SECONDARY;
    }
} elseif ( function_exists( 'amp_is_canonical' ) ) {
    return self::AMP_MODE_SECONDARY;
}
```

I know it seems a lot more verbose, but if new modes are added to the AMP plugin we should need to explicitly support them I'd think, which is what our old code is doing. Otherwise `if ( in_array( $mode, array( AMP_Theme_Support::TRANSITIONAL_MODE_SLUG, AMP_Theme_Support::READER_MODE_SLUG ), true ) ) {` would be skipped and we'd always `return self::AMP_MODE_SECONDARY`, but we don't.

So let's keep those checks in.

Reply:
Hi @tofumatt,
A couple of days ago Evan told us "develop has been cut into main so please ensure any remaining PRs for 1.35.0 (or follow-ups from QA) are created from and updated to target main.".  I took this to mean that I needed to create a new branch from main for this ticket, transfer the changes to it and create a new PR.  I guess this was the wrong approach.  What should I have done?

Reply:
@Hazlitte that comment was only regarding issues for the release (1.35.0). #2998 was not tagged with the release, so no change was needed here.

Reply:
@aaemnnosttv Ah ok, thanks for clarifying 👍